### PR TITLE
Better handle default port

### PR DESCRIPTION
**Pull request by florent-lamiraux**
When creating a client, environment variable HPP_PORT was not really
  taken into account. This commit fixes this issue and allows to control
  parallel instances of hppcorbaserver on different ports only by specifying
  variable HPP_PORT.

### CODE REVIEW

**Review by jmirabel — COMMENTED** [review]
Indeed, HPP_PORT was not used.

The code has many branches spread in several functions so it is a bit hard to follow. By merely looking at the code without testing, it looks good to me.

Maybe unit testing this could help documenting and ensuring the behavior.

**Review by jmirabel — APPROVED** [review]
I wanted to approve.

File: src/hpp/corbaserver/client.py [hunk 1]
```diff
@@ -132,10 +132,10 @@ def _initOrb(self, url, host=None, port=None):
                     self.initWithNameService(url + "/NameService")
         else:
             urlNameService = _getIIOPurl(
-                service="NameService", host=host, port=port if port else 2809
+                service="NameService", host=host, port=port, default_port=2809
             )
             urlHppTools = _getIIOPurl(
-                service="hpp-corbaserver", host=host, port=port if port else 13331
+                service="hpp-corbaserver", host=host, port=port, default_port=13331
```
Comment (nim65s) [inline src/hpp/corbaserver/client.py:138]:
```suggestion
                service="hpp-corbaserver", host=host, port=port if port else os.getenv("HPP_PORT", 13331)
```

maybe ? Otherwise I don't get the different use cases between this 13331 or the one you introduced in tools.py

In other words: If I ever want to change the default from beeing 13331 to 13332, should I change this `default_port=13331` here or the `port = os.getenv("HPP_PORT", 13331)` in tools.py ? and why would we have the 2 ?

**Review by nim65s — COMMENTED** [review]
I'm pretty sure a cast to int is missing around `os.getenv`. did you test it with a HPP_PORT defined ?

File: src/hpp/corbaserver/client.py (hunk 1 above)
Comment (florent-lamiraux) [reply src/hpp/corbaserver/client.py]: I defined defaultPort in class Client and use it in tools.py. Thank you for the remark.

**Comment by florent-lamiraux** [comment]
> I'm pretty sure a cast to int is missing around `os.getenv`. did you test it with a HPP_PORT defined ?

No need to cast into int since when only write the port in a string. Thus whatever the type, the result is the same.

**Comment by florent-lamiraux** [comment]
I also added a test on communication via port 13332.

File: tests/port.py [hunk 2]
```diff
@@ -0,0 +1,38 @@
...
+        time.sleep(3)
+        
+        client = Client(port=13332)
+        client.robot.createRobot("HRP-7")
+        self.assertEqual(client.robot.getRobotName(), "HRP-7")
+        os.environ['HPP_PORT'] = '13332'
+        client = Client()
+        self.assertEqual(client.robot.getRobotName(), "HRP-7")
+        self.process.terminate()
+        run(["killall", "hppcorbaserver"])
```
Comment (jmirabel) [inline tests/port.py:35]: For your information,
```python
tools = hpp.corbaserver.tools.Tools()
tools.shutdown()
```
should cleanly stop the server.

Comment (jmirabel) [reply tests/port.py:35]: Thanks for the unit test.
